corehq/apps/reports/tasks.py

```diff
@@ -145,14 +143,29 @@ def _send_email(user, report, hash_id, recipient):
     send_report_download_email(report.name, recipient, link)
 
 
-def _store_excel_in_redis(report_class, file):
-    hash_id = uuid.uuid4().hex
+def _store_excel_in_blobdb(report_class, file):
 
+    # Store classname to redis
+    key = uuid.uuid4().hex
     r = get_redis_client()
-    r.set(hash_id, [report_class, file.getvalue()])
-    r.expire(hash_id, EXPIRE_TIME)
-
-    return hash_id
+    r.set(key, report_class)
+    r.expire(key, EXPIRE_TIME)
+
+    # Store file to blobdb
+    expired = 60 * 60 * 24 * 7  # 7 days
+    db = get_blob_db()
+    parent_id = uuid.uuid4().hex
+
+    kw = {
+        "domain": DASHBOARD_DOMAIN,
```

Comment:
Was this only used for testing? It looks it will store the dashboard domain constant for all reports. I think you can pass in report.domain

Reply:
https://github.com/dimagi/commcare-hq/pull/23927/commits/33ac9a104cb6908dc9e726c314c68f3929f241e9